### PR TITLE
Add sharding and readwrite-splitting example conf for proxy.

### PR DESCRIPTION
**Pull request by totalo**
For #12366.

Changes proposed in this pull request:
- Add sharding and readwrite-splitting example conf for proxy.

### CODE REVIEW

File: examples/shardingsphere-proxy-example/shardingsphere-proxy-memory-example/shardingsphere-proxy-memory-local-sharding-example/pom.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.apache.shardingsphere.example</groupId>
+        <artifactId>shardingsphere-proxy-memory-example</artifactId>
+        <version>5.0.0-RC1-SNAPSHOT</version>
+    </parent>
+    <artifactId>shardingsphere-proxy-memory-example</artifactId>
+    <name>${project.artifactId}</name>
+    
+
```
Comment (terrymanu) [inline examples/shardingsphere-proxy-example/shardingsphere-proxy-memory-example/shardingsphere-proxy-memory-local-sharding-example/pom.xml:31]: Please remove useless empty lines

File: examples/shardingsphere-proxy-example/shardingsphere-proxy-memory-example/shardingsphere-proxy-memory-local-sharding-example/pom.xml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+</project>
```
Comment (terrymanu) [inline examples/shardingsphere-proxy-example/shardingsphere-proxy-memory-example/shardingsphere-proxy-memory-local-sharding-example/pom.xml:32]: Please add a new line in the end of file

**Review by terrymanu — CHANGES_REQUESTED** [review]

File: examples/shardingsphere-proxy-example/shardingsphere-proxy-memory-example/shardingsphere-proxy-memory-local-sharding-example/pom.xml (hunk 1 above)
Comment (totalo) [reply examples/shardingsphere-proxy-example/shardingsphere-proxy-memory-example/shardingsphere-proxy-memory-local-sharding-example/pom.xml]: Done, my negligence.

File: examples/shardingsphere-proxy-example/shardingsphere-proxy-memory-example/shardingsphere-proxy-memory-local-sharding-example/pom.xml (hunk 2 above)
Comment (totalo) [reply examples/shardingsphere-proxy-example/shardingsphere-proxy-memory-example/shardingsphere-proxy-memory-local-sharding-example/pom.xml]: Done, my negligence.

**Review by terrymanu — APPROVED** [review]

**Pull request merged**